core/integration/container_test.go

```diff
@@ -261,6 +261,24 @@ CMD echo "stage2"
 		require.Contains(t, output, "stage1\n")
 		require.NotContains(t, output, "stage2\n")
 	})
+
+	t.Run("with build secrets", func(t *testing.T) {
+		sec := c.SetSecret("my-secret", "barbar")
+		require.NoError(t, err)
+
+		src := contextDir.
+			WithNewFile("Dockerfile",
+				`FROM golang:1.18.2-alpine
+WORKDIR /src
+RUN --mount=type=secret,id=my-secret test "$(cat /run/secrets/my-secret)" = "barbar"
+RUN --mount=type=secret,id=my-secret cp /run/secrets/my-secret  /secret
+CMD cat /secret
```

Comment:
nit: improve this assertion to `CMD test $(cat /secret) == "barbar"`. 
We can then assert the go test with err == nil instead of "***"

Reply:
Since I already do this on line 273, I feel it's a repeat that would just confirm that the `cp` from the secret to another file works instead of also checking the scrubbing.

Reply:
> Since I already do this on line 273, I feel it's a repeat that would just confirm that the `cp` from the secret to another file works instead of also checking the scrubbing.

you're correct. Totally missed this.